api/filters/replacement/replacement_test.go

```diff
@@ -1431,6 +1431,136 @@ spec:
     name: second
     version: latest
     property: second`,
+		},
+		"index contains '*' character": {
+			input: `apiVersion: apps/v1
+kind: Deployment
+metadata:
+  labels:
+    app: sample-deploy
+  name: sample-deploy
+spec:
+  replicas: 1
+  selector:
+    matchLabels:
+      app: sample-deploy
+  template:
+    metadata:
+      labels:
+        app: sample-deploy
+    spec:
+      containers:
+      - image: nginx
+        name: main
+        env:
+        - name: deployment-name
+          value: XXXXX
+`,
+			replacements: `replacements:
+- source:
+    kind: Deployment
+    name: sample-deploy
+    fieldPath: metadata.name
+  targets:
+  - select:
+      kind: Deployment
+    fieldPaths:
+    - spec.template.spec.containers.*.env.[name=deployment-name].value
+`,
+			expected: `apiVersion: apps/v1
+kind: Deployment
+metadata:
+  labels:
+    app: sample-deploy
+  name: sample-deploy
+spec:
+  replicas: 1
+  selector:
+    matchLabels:
+      app: sample-deploy
+  template:
+    metadata:
+      labels:
+        app: sample-deploy
+    spec:
+      containers:
+      - image: nginx
+        name: main
+        env:
+        - name: deployment-name
+          value: sample-deploy`,
+		},
+		"list index contains '*' character": {
```

Comment:
what is the difference between this test and the above test?

Also, the test seems not to be passing.

Reply:
It tests to replace multi values.
- https://github.com/kubernetes-sigs/kustomize/pull/4424/files#diff-909c0819f13e268a60555f0aa78bd8acdbb42c10f3e2a906d62ba4b9d1722947R1515-R1522
- https://github.com/kubernetes-sigs/kustomize/pull/4424/files#diff-909c0819f13e268a60555f0aa78bd8acdbb42c10f3e2a906d62ba4b9d1722947R1556-R1563